streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -662,6 +664,16 @@ object StreamingContext extends Logging {
     }
   }
 
+  /**
+   * Get the context that is currently active, that is, started but not stopped.
+   */
+  @Experimental
+  def getActive(): Option[StreamingContext] = {
+    ACTIVATION_LOCK.synchronized {
+      Option(activeContext.get())
```

Comment:
I'm a bit confused about the locking here. What if I call start() then stop() currently on the same streamingContext in different threads. Can't I have a race condition where the stop() calls `setActiveContext(null)` between the following lines:

```
t1: state = StreamingContextState.ACTIVE
t2:   --> setActiveContext(null)
t1: setActiveContext(this)
```